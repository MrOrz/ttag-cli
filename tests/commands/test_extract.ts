import * as path from 'path';
import * as fs from 'fs';
import { execSync } from 'child_process';

const potPath = path.resolve(__dirname, '../../dist/translation.pot');
const baseTestPath = path.resolve(__dirname, '../fixtures/baseTest');

function cleanup() {
    fs.unlinkSync(potPath);
}

afterEach(() => {
  cleanup();
})

test('extract', () => {
  execSync(`ts-node src/index.ts extract -o ${potPath} ${baseTestPath}`);
  const result = fs.readFileSync(potPath).toString();
  expect(result).toMatchSnapshot()
});
