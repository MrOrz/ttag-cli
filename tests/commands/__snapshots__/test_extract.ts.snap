// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extract base case 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Content-Type: text/plain; charset=utf-8\\\\n\\"
\\"Plural-Forms: nplurals=2; plural=(n!=1);\\\\n\\"

#: tests/fixtures/baseTest/test1.js:4
msgid \\"test translation \${ name }\\"
msgstr \\"\\"

#: tests/fixtures/baseTest/nested/test2.js:4
msgid \\"test translation 2 \${ name }\\"
msgstr \\"\\""
`;

exports[`extract from jsx 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Content-Type: text/plain; charset=utf-8\\\\n\\"
\\"Plural-Forms: nplurals=2; plural=(n!=1);\\\\n\\"

#: tests/fixtures/testJSXParse.jsx:6
msgid \\"test from jsx\\"
msgstr \\"\\""
`;
